applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/mechanical_explicit_strategy.hpp

```diff
@@ -461,6 +486,103 @@ class MechanicalExplicitStrategy
+
+        const int number_of_constraints = static_cast<int>(rModelPart.MasterSlaveConstraints().size());
+
+        // Setting to zero the slave dofs
+        #pragma omp parallel firstprivate(slave_equation_dofs)
+        {
+            #pragma omp for schedule(guided, 512)
+            for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
+                auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
+
+                // Detect if the constraint is active or not. If the user did not make any choice the constraint
+                // It is active by default
+                bool constraint_is_active = true;
+                if (it_const->IsDefined(ACTIVE))
+                    constraint_is_active = it_const->Is(ACTIVE);
+
+                if (constraint_is_active) {
+                    slave_equation_dofs = it_const->GetSlaveDofsVector();
+
+                    for (IndexType i = 0; i < slave_equation_dofs.size(); ++i) {
+                        #pragma omp critical
+                        slave_equation_dofs[i]->GetSolutionStepValue() = 0.0;
+                    }
+                }
+            }
+        }
+
+        // Adding MPC contribution
+        #pragma omp parallel firstprivate(transformation_matrix, constant_vector, slave_equation_dofs, master_equation_dofs)
+        {
+            #pragma omp for schedule(guided, 512)
+            for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
+                auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
+
+                // Detect if the constraint is active or not. If the user did not make any choice the constraint
+                // It is active by default
+                bool constraint_is_active = true;
+                if (it_const->IsDefined(ACTIVE))
+                    constraint_is_active = it_const->Is(ACTIVE);
+
+                if (constraint_is_active) {
+                    it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
+                    slave_equation_dofs = it_const->GetSlaveDofsVector();
+                    master_equation_dofs = it_const->GetMasterDofsVector();
+
+                    // Resizing if needed
+                    if (slave_dofs_values.size() != slave_equation_dofs.size())
+                        slave_dofs_values.resize(slave_equation_dofs.size());
+
+                    if (master_dofs_values.size() != master_equation_dofs.size())
+                        master_dofs_values.resize(master_equation_dofs.size());
+
+                    for (IndexType i = 0; i < master_equation_dofs.size(); ++i) {
+                        master_dofs_values[i] = master_equation_dofs[i]->GetSolutionStepValue();
+                    }
+
+                    noalias(slave_dofs_values) = prod(transformation_matrix, master_dofs_values) + constant_vector;
```

Comment:
i advocate for adding a function Apply to the MasterSlaveconstraints so that all of the ops above can be done in there